### PR TITLE
🎉 Source Facebook Marketing: fix schema for 'breakdowns' fields

### LINKED ISSUE #8437: 🐛 Source Facebook Marketing: discover is broke when using valid breakdown value in a custom insight
## Environment
- **Airbyte version**:   0.32.8-alpha
- **OS Version / Instance**: example macOS, Windows 7/10, Ubuntu 18.04, GCP n2. , AWS EC2
- **Deployment**: example are Docker or Kubernetes deploy env
- **Source Connector and version**: Facebook Marketing 0.2.26
- **Severity**: Critical
- **Step where error happened**: Discover schema

## Current Behavior
I reproduced a user's bug ([slack conversation](https://airbytehq.slack.com/archives/C01MFR03D5W/p1638397104248300)): when I set a `custom_insights` in the connection with a breakdown value such as `product_id` the connection test is sucessful but the schema discovery that follows crashes.

This is due to the fact that the list in the `schemas` dict https://github.com/airbytehq/airbyte/blob/9f685b95c87d3af1571594d1cd568943ec567daa/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py#L421
 does not list all valid values for this field. The list used by the connection check (all the properties of [`ads_insights_breakdowns.json`](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/ads_insights_breakdowns.json) looks more accurate
## Expected Behavior
* Successful discovery with all the possible breakdown value supported by Facebook marketing API.
* A single source of through for all the valid values the `breakdown` field can take.
* 
## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```

2021-12-02 14:58:41 INFO () DockerProcessFactory(create):127 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/0d25e941-5540-4141-82bf-d12d139558e7/0 --network host --log-driver none airbyte/source-facebook-marketing:0.2.26 discover --config source_config.json
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 - Traceback (most recent call last):
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -   File "/airbyte/integration_code/main.py", line 13, in <module>
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -     launch(source, sys.argv[1:])
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/entrypoint.py", line 108, in launch
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -     for message in source_entrypoint.run(parsed_args):
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/entrypoint.py", line 93, in run
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -     catalog = self.source.discover(self.logger, config)
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 68, in discover
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -     streams = [stream.as_airbyte_stream() for stream in self.streams(config=config)]
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 68, in <listcomp>
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -     streams = [stream.as_airbyte_stream() for stream in self.streams(config=config)]
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/streams/core.py", line 66, in as_airbyte_stream
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -     stream = AirbyteStream(name=self.name, json_schema=dict(self.get_json_schema()), supported_sync_modes=[SyncMode.full_refresh])
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -   File "/airbyte/integration_code/source_facebook_marketing/streams.py", line 408, in get_json_schema
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -     schema["properties"].update(self._schema_for_breakdowns())
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -   File "/airbyte/integration_code/source_facebook_marketing/streams.py", line 436, in _schema_for_breakdowns
2021-12-02 14:58:43 ERROR () LineGobbler(voidCall):82 -     return {breakdown: schemas[breakdown] for breakdown in self.breakdowns}

```

</details>

## Steps to Reproduce
1. Set up a Facebook Marketing connection with custom insights using `product_id` in breakdown
2. Connection test passes
3. On a new connection setup between source and destination feching schema fails

Screen Shot 2021-12-02 at 16 07 58

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
Bugfix schema generation for  `AdsInsights`stream and it's child streams.

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [x] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [x] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml [hunk 1]
```diff
@@ -169,7 +169,7 @@
 - name: Facebook Marketing
   sourceDefinitionId: e7778cfc-e97c-4458-9ecb-b4f2bba8946c
   dockerRepository: airbyte/source-facebook-marketing
-  dockerImageTag: 0.2.26
+  dockerImageTag: 0.2.27
```
Comment (keu) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:172]: this needs to be updated after publish only

**Review by keu — APPROVED** [review]
just small comment, LGTM otherwise

File: docs/integrations/sources/facebook-marketing.md [hunk 2]
```diff
@@ -96,6 +96,7 @@ As a summary, custom insights allows to replicate only some fields, resulting in
 
 | Version | Date | Pull Request | Subject |
 | :--- | :--- | :--- | :--- |
+| 0.2.27  | 2021-12-13 | [8742](https://github.com/airbytehq/airbyte/pull/8742) | Bufix schema generation related to "breakdown" fields |
```
Comment (bazarnov) [inline docs/integrations/sources/facebook-marketing.md:99]: nit: Fix for

**Review by bazarnov — APPROVED** [review]

File: docs/integrations/sources/facebook-marketing.md (hunk 2 above)
Comment (grubberr) [reply docs/integrations/sources/facebook-marketing.md]: changed

**Comment by sherifnada** [comment]
@grubberr airbyte review not needed as it doesn't affect ux, internal is sufficient

**Review by antixar — APPROVED** [review]

**Pull request merged**